### PR TITLE
[DO NOT MERGE] Overview of removing problematic whitelist/blacklist terminology

### LINKED ISSUE #43302: Replace "whitelist" and "blacklist" with "allowed" and "disallowed"
There are a number of places in our repository which use the terms "whitelist" and "blacklist" unnecessarily and which are under our control. In the context of our current social structures, the terms "white" and "black" are not neutral, and using "whitelist" and "blacklist" closely mirrors racial associations of whiteness or the color white being or representing something positive and blackness or the color back being or representing something negative. Using these terms in this way perpetuates racism ingrained in the English language through centuries of colonialism and racism.

Additionally, alternative language to "whitelist" and "blacklist" will always be generally more clear.

Given that in every case under our control it should be possible to come up with alternative language that is not weighed with racial connotations, we should do this in our repository.

A quick search of our repository yielded around 65 files that used the term "whitelist". Some files, like stylelint, webpack, and eslint configurations and yarn releases I don't think we would be able to have control over unless the underlying libraries were to be updated.

There are about 20 files which use the term "blacklist".

Under most circumstances, "whitelist" can be substituted with "allowed" and "blacklist" with "disallowed". In the handful of cases when that's not possible (Grammatical issues in user-displayed copy for example—though I'm not certain these issue will exist, I'm anticipating they might) then it should not be difficult to come up with an appropriate alternative.

### PR DESCRIPTION
**Pull request by sarayourfriend**
#### Changes proposed in this Pull Request

* Change instances of the words "whitelist" and "blacklist" to alternative (often times more descriptive) terms https://github.com/Automattic/wp-calypso/issues/43302

#### Testing instructions

* All unit tests should pass
* todo: potentially manual testing is necessary

Fixes https://github.com/Automattic/wp-calypso/issues/43302

### CODE REVIEW

File: client/lib/plugins/utils.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ function filterNoticesBy( site, pluginSlug, log ) {
 	return false;
 }
 
+// @todo(saramarcondes) Is this safe to rename? Would plugins potentially depend on this function's current name?
```
Comment (saramarcondes) [inline client/lib/plugins/utils.js:53]: @kwight I noticed you made the most recent "substantial" (non-formatting/chore) change to this file. Do you happen to know the answer to my question in this comment? Is it safe to rename this `whiteListPluginData` function?

Comment (kwight) [reply client/lib/plugins/utils.js:53]: From what I can tell, the only two uses are here in this file: the definition, and a single call. The tests will have to be updated, but other than that, I think it's good to go!

Comment (saramarcondes) [reply client/lib/plugins/utils.js:53]: Right on, thanks so much!

File: client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ const SiteTitle: React.FunctionComponent< Props > = ( { onSubmit } ) => {
 					spellCheck={ false }
 					autoComplete="off"
 					autoCorrect="off"
-					data-hj-whitelist
+					data-hj-whitelist // @todo(saramarcondes) where is this used?
```
Comment (simison) [inline client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx:53]: This is to allow Hotjar.com show what user types into this field and unfortunately we cannot modify it.

https://help.hotjar.com/hc/en-us/articles/115015563287-How-to-Whitelist-Elements-and-Keystrokes

Comment (saramarcondes) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx:53]: Ah okay, thanks for letting me know. I’ll add a note in a comment!

File: client/my-sites/site-settings/form-discussion.jsx [hunk 3]
```diff
@@ -525,22 +525,22 @@ class SiteSettingsFormDiscussion extends Component {
...
+					name="disallowed_comment_keys"
+					id="disallowed_comment_keys"
 					value={ fields.blacklist_keys }
 					onChange={ onChangeField( 'blacklist_keys' ) }
 					disabled={ isRequestingSettings || isSavingSettings }
 					autoCapitalize="none"
-					onClick={ eventTracker( 'Clicked Blacklist Field' ) }
-					onKeyPress={ uniqueEventTracker( 'Typed in Blacklist Field' ) }
+					onClick={ eventTracker( 'Clicked Disallowed Comments Field' ) }
+					onKeyPress={ uniqueEventTracker( 'Typed in Disallowed Comments Field' ) }
```
Comment (saramarcondes) [inline client/my-sites/site-settings/form-discussion.jsx:543]: Discussion about whether this is safe: p5uIfZ-9OH-p2 (it sounds like it is!)

File: client/lib/domains/registration/availability-messages.js [hunk 4]
```diff
@@ -254,7 +254,7 @@ function getAvailabilityNotice( domain, error, errorData ) {
 				);
 			} else {
 				message = translate(
-					'Domain cannot be mapped to a WordPress.com blog because of blacklisted term.'
+					'Domain cannot be mapped to a WordPress.com blog because of disallowed term.'
```
Comment (a8ci18n) [inline client/lib/domains/registration/availability-messages.js:257]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

**Comment by saramarcondes** [comment]
The JS lint job is failing due to there being too high a number of JS files being passed to `eslint` in the `FILES_TO_LINT` variable. Not sure what to do about that but I've opened a ticket for Team Calypso to address the issue: https://github.com/Automattic/wp-calypso/issues/43631

File: client/blocks/reader-feed-header/index.jsx [hunk 5]
```diff
@@ -13,7 +13,7 @@ import config from 'config';
  */
 import { Card } from '@automattic/components';
 import ReaderFollowButton from 'reader/follow-button';
-import { isAuthorNameBlocked } from 'reader/lib/author-name-blocklist';
```
Comment (gwwar) [inline client/blocks/reader-feed-header/index.jsx:16]: Is blocklist troublesome here?

File: client/jetpack-connect/connection-notice-types.js [hunk 6]
```diff
@@ -26,7 +26,7 @@ export const INVALID_PERMISSIONS = 'invalidPermissions';
 export const RETRY_AUTH = 'retryAuth';
 export const RETRYING_AUTH = 'retryingAuth';
 export const SECRET_EXPIRED = 'secretExpired';
-export const SITE_BLACKLISTED = 'siteBlacklisted';
+export const SITE_BLOCKED = 'siteBlacklisted';
```
Comment (gwwar) [inline client/jetpack-connect/connection-notice-types.js:29]:
```suggestion
export const SITE_BLOCKED = 'siteBlocked';
```

**Review by gwwar — COMMENTED** [review]
If we'd like manual testing, I'd recommend splitting this out to a few PRs. Maybe one for Jetpack notices, and the rest a reasonable number of files per a certain set of folders/functionality.

File: client/blocks/reader-feed-header/index.jsx (hunk 5 above)
Comment (saramarcondes) [reply client/blocks/reader-feed-header/index.jsx:16]: I suppose not, it's probably one of the cases where `blocklist` is pretty clear actually. I'll revert this one.

**Comment by saramarcondes** [comment]
@gwwar Thanks for the review and suggestions! I'd definitely prefer manual testing. Agreed that splitting it makes sense for that. I'll get split PRs open either today or tomorrow.

File: client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx (hunk 2 above)
Comment (saramarcondes) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx]: I reached out to HotJar yesterday and got a response from them that they're working on this internally. They're going to let me know once the data attribute has been updated to be able to be changed to something different.

Comment (simison) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx]: Excellent! Thanks for checking with them.

Comment (p-jackson) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx]: Looks like the new attribute name is `data-hj-allow`.
https://help.hotjar.com/hc/en-us/articles/115015563287-How-to-Whitelist-Elements-and-Keystrokes

Updating eslint exception here: #46620

**Pull request closed without merging**